src/components/common/YourCompany/StartDate.vue

```diff
@@ -0,0 +1,214 @@
+          </v-btn>
+          <v-menu v-if="isCorrectionFiling && !onEditMode && isCorrected"
+            offset-y left nudge-bottom="4" v-model="dropdown">
+            <template v-slot:activator="{ on }">
+              <v-btn text small color="primary" id="start-menu-btn" v-on="on">
+                <v-icon>{{dropdown ? 'mdi-menu-up' : 'mdi-menu-down'}}</v-icon>
+              </v-btn>
+            </template>
+            <v-btn text color="primary" id="more-changes-btn"
+              @click="onChangeClicked(); dropdown = false">
+              <v-icon small color="primary">mdi-pencil</v-icon>
+              <span>{{ editLabel }}</span>
+            </v-btn>
+          </v-menu>
+        </span>
+
+      </v-col>
+    </v-row>
+  </section>
+</template>
+
+<script lang="ts">
+import { Component, Mixins, Prop, Watch } from 'vue-property-decorator'
+import { Getter, Action } from 'vuex-class'
+import { CommonMixin, DateMixin } from '@/mixins/'
+import { DatePicker as DatePickerShared } from '@bcrs-shared-components/date-picker'
+import { ActionBindingIF } from '@/interfaces'
+
+@Component({
+  components: {
+    DatePickerShared
+  }
+})
+export default class StartDate extends Mixins(CommonMixin, DateMixin) {
+  // Global getters
+  @Getter isBenIaCorrectionFiling!: boolean
+  @Getter isFirmCorrectionFiling!: boolean
+  @Getter getCorrectedFilingDate!: string
+  @Getter getBusinessFoundingDate!: string // actually date-time
+  @Getter getCurrentDate!: string
+  @Getter getCurrentJsDate!: Date
+
+  // Global setter
+  @Action setCorrectionFilingDate!: ActionBindingIF
+  @Action setValidComponent!: ActionBindingIF
+
+  /** Whether to show invalid section styling. */
+  @Prop({ default: false })
+  readonly invalidSection: boolean
+
+  protected onEditMode = false
+  protected isCorrected = !!this.getCorrectedFilingDate
+  protected newCorrectedStartDate = undefined // date is "Month Day, Year"
+
+  /** The minimum start date that can be entered (Up to 2 years ago today). */
+  protected get startDateMin (): Date {
+    const startDateMin = new Date(this.getCurrentJsDate)
+    startDateMin.setFullYear(startDateMin.getUTCFullYear() - 2)
+    startDateMin.setHours(0, 0, 0) // Set time to 0 for accurate Date Rules comparison
+
+    return startDateMin
+  }
+
+  /** The minimum start date string. */
+  protected get startDateMinStr (): string {
+    return this.dateToYyyyMmDd(this.startDateMin)
+  }
+
+  /** The maximum start date that can be entered (Up to 90 days from today). */
+  protected get startDateMax (): Date {
+    const startDateMax = new Date(this.getCurrentJsDate)
+    startDateMax.setDate(startDateMax.getUTCDate() + 90)
+    return startDateMax
+  }
+
+  /** The maximum start date string. */
+  protected get startDateMaxStr (): string {
+    return this.dateToYyyyMmDd(this.startDateMax)
+  }
+
+  /** The recognition date or business start date-time string. */
+  protected get recognitionDateTime (): string {
```

Comment:
Some filings display "recognition date and time" and other display "business start date". Does this new component work with the 2 types?

As far as I know:
- BC alteration: recognition date and time
- BC correction: recognition date and time
- firm correction: business start date
- firm change: business start date
- firm conversion: business start date

Reply:
I am not actually sure I will have to verify this

Reply:
I think I erred above. THIS component is for firm correction/change/conversion filings (business start date). There's a separate section for BC alteration/correction.

Reply:
This does mean some of your code in here doesn't need to handle those other filings.

Or I'm really confused and you're trying to handle both cases in this component. (But if so, it's incomplete).

Reply:
Okay I will clean up this section after I verify what filing types are using this component

Reply:
Did you test all 4 scenarios above?

Reply:
Yes I am now testing with all 5 of the scenarios above.